### DISCUSSION TITLE
Como fazer um link que abre uma busca usando o buscador padrão do browser?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#1776] [Perguntas e Respostas]
Olá pessoal, estou fazendo algumas páginas em que terá links que abrem uma nova página com uma busca no Google, porém... e se o buscador padrão do browser for o bing ou o duckduckgo?

Encontrei alguns [URL Schemes reservados no Windows](https://docs.microsoft.com/en-us/windows/uwp/launch-resume/reserved-uri-scheme-names#reserved-uri-scheme-names), mas nada que seja o que estou procurando.

Também encontrei https://www.w3.org/wiki/UriSchemes que lista vários protocolos exóticos para mim.

Porém nada como um hipotético `search-for-this://`, o mais próximo que ouvi as pessoas falando foi o `x-web-search://` que supostamente funciona para [buscar algo no iOS Safari](https://gist.github.com/felquis/a08ee196747f71689dcb#gistcomment-1877158).

Alguém tem alguma ideia?

**Comment by wilsonneto-swe** [comment]
Sendo sincero não sei responder a pergunta em questão, 
mas diria "Porque não usa apenas o goole?" Se quer buscar algo no google apenas mude a URL: "https://www.google.com.br/search?q=azure145" (o parametro "q" é a busca). 

Acho mais válido usar o básico do que usar algum protocolo exótico que funcione em uma ou duas versões de browsers e quebre nos demais.

Mas se descobrir algo funcional, compartilha ai hehe
abraço!

**Comment by felquis** [comment]
Não quero simplesmente assumir que o planeta todo uso o Google para buscar, mesmo que seja gigante e ótimo. A solução simples e direta ao ponto que é usar o google eu já fiz.

Porém, tem sim uns loucos que usam Bing, Baidu, DuckGo, e sabe Deus o que mais. Não quero que essas pessoas tenham que ser obrigados a ver um resultado de busca no google.

Por este motivo entrei nessa questão bizarra que eu nunca tinha me perguntando antes.

**Comment by portothree** [comment]
Dentro do mozilla XPCOM `nsIBrowserSearchService` existe um method `getDefaultEngines()` que retorna uma array com os search engines escolhidos por defeito e pode ser utilizado para extensions pra firefox. Acredito que neste caso ficaria obrigado a buscar uma soluçao para cada browser...

O que faria é um tooltip com 3-4 search engines mais famosos e o user pode escolher em qual fazer a busca.
